plugins/modules/win_route.ps1

```diff
@@ -10,95 +11,120 @@
-      $result.changed = $true
-      $result.output = "Route added"
+    # Check if the static route is already present
+    $Route = Get-CimInstance -ClassName Win32_IP4PersistedRouteTable | Where-Object { $_.Destination -EQ "$($Network)" -AND $_.Mask -EQ "$($Mask)" }
 
+    # In case of running in Check Mode
+    if ($CheckMode -AND !($Route)) {
+        $result.changed = $true
+        $result.output = "Route can be added!"
+        return
     }
-    catch {
-      $ErrorMessage = $_.Exception.Message
-      Fail-Json $result $ErrorMessage
-    }
-  }
-  else {
-    $result.output = "Static route already exists"
-  }
 
+    # If route doesn't exist, add it
+    if (!($Route)) {
+        try {
+            # Add static route
+            route -p ADD $Network MASK $Mask $Gateway METRIC $Metric
+            $result.changed = $true
+            $result.output = "Route added!"
+        }
+        catch {
+            $ErrorMessage = $_.Exception.Message
+            Fail-Json $result $ErrorMessage
+        }
+    }
+    # If route already exists, just print
+    else {
+        $result.output = "Static route already exists."
+    }
 }
 
+###########################
+### Remove static route ###
+###########################
+
 Function Remove-Route {
-  Param (
-    [Parameter(Mandatory=$true)]
-    [string]$Destination,
-    [bool]$CheckMode
-    )
-  $IpAddress = $Destination.split('/')[0]
-  $Route = Get-CimInstance win32_ip4PersistedrouteTable -Filter "Destination = '$($IpAddress)'"
-  if ($Route){
-    try {
-
-      Remove-NetRoute -DestinationPrefix $Destination -Confirm:$false -ErrorAction Stop -WhatIf:$CheckMode
-      $result.changed = $true
-      $result.output = "Route removed"
-    }
-    catch {
-      $ErrorMessage = $_.Exception.Message
-      Fail-Json $result $ErrorMessage
-    }
-  }
-  else {
-    $result.output = "No route to remove"
-  }
+    Param (
+        [Parameter(Mandatory=$true)]
+        [string]$Network,
 
-}
+        [Parameter(Mandatory=$true)]
+        [string]$Mask,
 
-# Set gateway if null
-if(!($gateway)){
-  $gateway = "0.0.0.0"
-}
+        [Parameter(Mandatory=$true)]
+        [string]$Gateway,
 
+        [Parameter(Mandatory=$true)]
+        [bool]$CheckMode
+    )
 
-if ($state -eq "present"){
+    # Check if the static route is already present
+    $Route = Get-CimInstance -ClassName Win32_IP4PersistedRouteTable | Where-Object { $_.Destination -EQ "$($Network)" -AND $_.Mask -EQ "$($Mask)" }
+
+    # In case of running in Check Mode
+    if ($CheckMode -AND $Route) {
+        $result.changed = $true
+        $result.output = "Route can be removed!"
+        return
+    }
 
-  Add-Route -Destination $dest -Gateway $gateway -Metric $metric -CheckMode $check_mode
+    # If route exists, remove it
+    if ($Route) {
+        try {
+            # Remove static route
+            route DELETE $Network MASK $Mask $Gateway
```

Comment:
The trouble with moving to a native command from a cmdlet is that error processing is a whole different beast with PowerShell. There are so many different behaviour permutations when it comes to detecting errors and formatting the output that I'm never confident this will actually catch an error.